core/src/eval/mod.rs

```diff
@@ -960,7 +965,11 @@ pub fn subst<C: Cache>(
         | v @ Term::SealingKey(_)
         | v @ Term::Enum(_)
         | v @ Term::Import(_)
-        | v @ Term::ResolvedImport(_) => RichTerm::new(v, pos),
+        | v @ Term::ResolvedImport(_)
+        // We could recurse here, because types can contain terms which would then be subject to
+        // substitution. Not recursing should be fine, though, because a type in term position
+        // turns into a contract, and we don't serialize contracts.
```

Comment:
Just in case, because substitution isn't only used for serialization, I believe (for example to pretty print a result in the REPL)

```suggestion
        // turns into a contract, and we don't substitute inside contracts either currently.
```